---
title: "Results for GCST90012877_buildGRCh37"
author: "XSun"
date: "2023-04-18"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r echo=FALSE, message=FALSE, warning=FALSE}
suppressMessages(library(data.table))
suppressMessages(library(ggplot2))
```

# Introduction

The GWAS data here is GCST90012877_buildGRCh37

# Results

## Genes and SNPs with susie pip > 0.8

The results are saved in "/project2/guiming/xsun/1.brain_wingo/results/877/877.susieIrss.txt"

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=5}

data <- fread("/project2/guiming/xsun/1.brain_wingo/results/877/877.susieIrss.txt")
data_pip08 <- data[data$susie_pip >0.8,][, c("chrom", "id", "pos", "type", "susie_pip")]

print("Genes and SNPs with susie pip >0.8")
DT::datatable(data_pip08)
```

## Comparing with Wingo's results 

Comparing with AD1 from Suppl Table 11 [Wingo_Nat_comm_2022](https://www.nature.com/articles/s41467-022-31873-5s)

```{r echo=FALSE, message=FALSE, warning=FALSE,fig.height=5, fig.width=5}

data_our <- data_pip08[data_pip08$type =="gene",]
data_wingo <- read.table("/project2/guiming/xsun/1.brain_wingo/data/AD1_wingo.txt",header = T)

print("overlapped")
data_overlap <- merge(data_our, data_wingo, by.x="id", by.y = "Ensembl_gene_ID")
DT::datatable(data_overlap)

print("our unique genes")
data_our_uniq <- data_our[!(data_our$id%in%data_wingo$Ensembl_gene_ID),]
DT::datatable(data_our_uniq)

print("wingo's unique genes")
data_wingo_uniq <- data_wingo[!(data_wingo$Ensembl_gene_ID%in%data_our$id),]
DT::datatable(data_wingo_uniq)
```