---
title: "Results for GCST90012877_buildGRCh37 -- latest ctwas package (LD REF: UKBB, Weights: fusion)"
author: "XSun"
date: "2023-06-10"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r echo=FALSE, message=FALSE, warning=FALSE }
results_dir <- paste0("/project2/guiming/xsun/proteome_alzheimer/2.brain_wingo_latest_fusionwgts_ukbb/brain_output/")
source("/project2/guiming/xsun/proteome_alzheimer/codes/ctwas_config_b37.R")
options(digits = 4)
```


# Weight QC
```{r echo=FALSE, message=FALSE, warning=FALSE }
# load Weight QC results
qclist_all <- readRDS(paste0(results_dir, "/weight_QC.RDS"))

#load information for all genes
files_weights <- list.files("/project2/guiming/data_download/Wingo_NC_2022/shared_mechanism.pQTLs.fusion.WEIGHTS_unzip/SampleID-map.nonNAfam.fusion.WEIGHTS/train_weights")

#number of weights in database
print("number of weights in database")
length(files_weights)

#number of imputed weights
print("number of imputed weights")
nrow(qclist_all)

#number of imputed weights by chromosome
print("number of imputed weights by chromosome")
table(qclist_all$chr)

#proportion of imputed weights without missing variants
print("proportion of imputed weights without missing variants")
mean(qclist_all$nmiss==0)
```


# ctwas results

```{r echo=FALSE, message=FALSE, warning=FALSE }

ctwas_gene_res <- readRDS(paste0(results_dir,"/brain_ctwas_gene_res.RDS"))
ctwas_res <- readRDS(paste0(results_dir, "/processed_gene&snp.RDS"))

load(paste0(results_dir, "/brain_expr_z_snp.Rd"))

sample_size <- z_snp$ss
sample_size <- as.numeric(names(which.max(table(sample_size))))

ctwas_res_s1 <- data.table::fread(paste0(results_dir, "/brain_ctwas.s1.susieIrss.txt"))
n_snps <- sum(ctwas_res_s1$type=="SNP")/thin
rm(ctwas_res_s1)

```


## Check convergence of parameters

```{r echo=FALSE, message=FALSE, warning=FALSE }
library(ggplot2)
library(cowplot)

load(paste0(results_dir, "/brain_ctwas.s2.susieIrssres.Rd"))

#estimated group prior (all iterations)
estimated_group_prior_all <- group_prior_rec
estimated_group_prior_all["SNP",] <- estimated_group_prior_all["SNP",]*thin #adjust parameter to account for thin argument

#estimated group prior variance (all iterations)
estimated_group_prior_var_all <- group_prior_var_rec

#set group size
group_size <- c(table(ctwas_gene_res$type), structure(n_snps, names="SNP"))
group_size <- group_size[rownames(estimated_group_prior_all)]

#estimated group PVE (all iterations)
estimated_group_pve_all <- estimated_group_prior_var_all*estimated_group_prior_all*group_size/sample_size #check PVE calculation

#estimated enrichment of genes (all iterations)
estimated_enrichment_all <- t(sapply(rownames(estimated_group_prior_all)[rownames(estimated_group_prior_all)!="SNP"], function(x){estimated_group_prior_all[rownames(estimated_group_prior_all)==x,]/estimated_group_prior_all[rownames(estimated_group_prior_all)=="SNP"]}))

title_size <- 12

df <- data.frame(niter = rep(1:ncol(estimated_group_prior_all), nrow(estimated_group_prior_all)),
                 value = unlist(lapply(1:nrow(estimated_group_prior_all), function(x){estimated_group_prior_all[x,]})),
                 group = rep(rownames(estimated_group_prior_all), each=ncol(estimated_group_prior_all)))

df$group[df$group=="Liver"] <- "Liver_Expression"
df$group <- as.factor(df$group)

p_pi <- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab("Iteration") + ylab(bquote(pi)) +
  ggtitle("Proportion Causal") +
  theme_cowplot()

p_pi <- p_pi + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = "Group")) + theme (legend.title = element_text(size=12, face="bold"))

df <- data.frame(niter = rep(1:ncol(estimated_group_prior_var_all), nrow(estimated_group_prior_var_all)),
                 value = unlist(lapply(1:nrow(estimated_group_prior_var_all), function(x){estimated_group_prior_var_all[x,]})),
                 group = rep(rownames(estimated_group_prior_var_all), each=ncol(estimated_group_prior_var_all)))

df$group[df$group=="Liver"] <- "Liver_Expression"
df$group <- as.factor(df$group)

p_sigma2 <- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab("Iteration") + ylab(bquote(sigma^2)) +
  ggtitle("Effect Size") +
  theme_cowplot()

p_sigma2 <- p_sigma2 + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = "Group")) + theme (legend.title = element_text(size=12, face="bold"))

df <- data.frame(niter = rep(1:ncol(estimated_group_pve_all), nrow(estimated_group_pve_all)),
                 value = unlist(lapply(1:nrow(estimated_group_pve_all), function(x){estimated_group_pve_all[x,]})),
                 group = rep(rownames(estimated_group_pve_all), each=ncol(estimated_group_pve_all)))

df$group[df$group=="Liver"] <- "Liver_Expression"
df$group <- as.factor(df$group)

p_pve <- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab("Iteration") + ylab(bquote(h^2[G])) +
  ggtitle("PVE") +
  theme_cowplot()

p_pve <- p_pve + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = "Group")) + theme (legend.title = element_text(size=12, face="bold"))

df <- data.frame(niter = rep(1:ncol(estimated_enrichment_all), nrow(estimated_enrichment_all)),
                 value = unlist(lapply(1:nrow(estimated_enrichment_all), function(x){estimated_enrichment_all[x,]})),
                 group = rep(rownames(estimated_enrichment_all), each=ncol(estimated_enrichment_all)))

df$group[df$group=="Liver"] <- "Liver_Expression"
df$group <- as.factor(df$group)

p_enrich <- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab("Iteration") + ylab(bquote(pi[G]/pi[S])) +
  ggtitle("Enrichment") +
  theme_cowplot()

p_enrich <- p_enrich + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = "Group")) + theme (legend.title = element_text(size=12, face="bold"))

plot_grid(p_pi, p_sigma2, p_enrich, p_pve)
```


```{r  message=FALSE, warning=FALSE }
#estimated group prior
estimated_group_prior <- estimated_group_prior_all[,ncol(group_prior_rec)]
print(estimated_group_prior)

#estimated group prior variance
estimated_group_prior_var <- estimated_group_prior_var_all[,ncol(group_prior_var_rec)]
print(estimated_group_prior_var)

#estimated enrichment
estimated_enrichment <- estimated_enrichment_all[ncol(group_prior_var_rec)]
print(estimated_enrichment)

#report sample size
print(sample_size)

#report group size
print(group_size)

#estimated group PVE
estimated_group_pve <- estimated_group_pve_all[,ncol(group_prior_rec)]
print(estimated_group_pve)

#total PVE
sum(estimated_group_pve)

#attributable PVE
estimated_group_pve/sum(estimated_group_pve)
```

## Genes with high pip

```{r echo=FALSE, message=FALSE, warning=FALSE }
#distribution of PIPs
hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main="Distribution of Gene PIPs")

print("Genes with susie pip >0.8")
ctwas_gene_respip08 <- ctwas_gene_res[ctwas_gene_res$susie_pip > 0.8,]
DT::datatable(ctwas_gene_respip08[order(-ctwas_gene_respip08$susie_pip),])
```

# Comparing with Wingo's results 

## Detailed results

Comparing with AD1 from Suppl Table 4 [Wingo_Nat_comm_2022](https://www.nature.com/articles/s41467-022-31873-5s)

```{r echo=FALSE, message=FALSE, warning=FALSE}

data_our <- ctwas_gene_respip08
id <- unlist(strsplit(data_our$id,split = "[.]"))[seq(1,2*nrow(data_our), by=2)]
data_our$genename <- unlist(strsplit(data_our$id,split = "[.]"))[seq(2,2*nrow(data_our), by=2)]
data_our$id <- id

data_wingo <- read.table("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo.txt",header = T)

print("overlapped")
data_overlap <- merge(data_our, data_wingo, by.x="id", by.y = "Ensembl_gene_ID")
DT::datatable(data_overlap)

print("our unique genes")
data_our_uniq <- data_our[!(data_our$id%in%data_wingo$Ensembl_gene_ID),]
DT::datatable(data_our_uniq)

print("wingo's unique genes")
data_wingo_uniq <- data_wingo[!(data_wingo$Ensembl_gene_ID%in%data_our$id),]
DT::datatable(data_wingo_uniq)
```



## Plots for the overlapped genes

```{r message=FALSE, warning=FALSE }
source("/project2/guiming/xsun/proteome_alzheimer/codes/locus_plot_fusion.R")
analysis_id <- "brain"
load("/project2/guiming/xsun/proteome_alzheimer/data_others/G_list_processed.RData")
alpha <- 0.05
ctwas_res$group <- "Expression"
ctwas_res[ctwas_res$type=="SNP",]$group <- "SNP"

a <- locus_plot(region_tag="6_36", return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel="both",
                      legend_side="left",
                      legend_panel="")

a <- locus_plot(region_tag="16_24", return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel="both",
                      legend_side="left",
                      legend_panel="")


a <- locus_plot(region_tag="17_37", return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel="both",
                      legend_side="left",
                      legend_panel="")


a <- locus_plot(region_tag="10_39", return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel="both",
                      legend_side="left",
                      legend_panel="")


a <- locus_plot(region_tag="2_25", return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel="both",
                      legend_side="left",
                      legend_panel="")

a <- locus_plot(region_tag="15_27", return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel="both",
                      legend_side="left",
                      legend_panel="")
```