---
title: "Jansen's GWAS (Merge = F)"
author: "XSun"
date: "2024-02-04"
output: 
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Overview

## Traits

Full GWAS (/project2/guiming/xsun/proteome_alzheimer/data_gwas/jansen.RDS) and top locus removed GWAS  (/project2/guiming/xsun/proteome_alzheimer/data_gwas/jansen.topremoved.RDS)

## Settings

1. preharmonize snp z score: harmonize_z=T, strand_ambig_action_z=“drop”

2. impute gene z-scores for both sets of prediction weights by chromosome harmonize_z = F (we have pre-harmonized) harmonize_wgt = T strand_ambig_action_wgt=“drop”;  scale_by_ld_variance=F (since the weight file was converted from FUSION format) 

3. ctwas_rss parameter estimation & fine mapping LD merge=F, group_prior_var_structure = “idependent”

4. Resourse requested: single qtl : --cpus-per-task=5 --mem=56G; joint analysis: --cpus-per-task=5 --mem=56G

## Functions used

```{r message=FALSE}
library(ctwas)
library(data.table)
library(DT)
library(dplyr)

thin <- 0.1
load("/project2/guiming/xsun/proteome_alzheimer/data_others/G_list_processed.RData")
G_list <- G_list[!duplicated(G_list$hgnc_symbol),]


process_para <- function(ctwas_parameters) {
  
  para <- cbind(ctwas_parameters$group_size,
                  ctwas_parameters$group_prior,
                  ctwas_parameters$group_prior_var,
                  c(NA, ctwas_parameters$enrichment),
                  ctwas_parameters$group_pve)
  para <- as.data.frame(cbind(rownames(para), para))
  colnames(para) <- c("group","group_size","group_prior","group_prior_var","enrichment","group_pve")
  return(para)
}

process_data <- function(weight, outputdir, outname, z_gene, display_datatable = TRUE) {
  
  if(!exists("G_list")) {
    load("/project2/guiming/xsun/proteome_alzheimer/data_others/G_list_processed.RData")
  }
  
  # Splitting and constructing weight file paths
  #weight <- unlist(strsplit(weight, split=","))
  #weight <- paste0("/project2/guiming/xsun/proteome_alzheimer/data_weights/", weight, ".db")

  # Reading cTWAS results
  ctwas_res <- fread(paste0(outputdir, outname, ".susieIrss.txt"))

  #separate gene and SNP results
  ctwas_gene_res <- ctwas_res[ctwas_res$type != "SNP", ]
  ctwas_gene_res <- data.frame(ctwas_gene_res)
  #####
  # ctwas_snp_res <- ctwas_res[ctwas_res$type == "SNP", ]
  # ctwas_snp_res <- data.frame(ctwas_snp_res)
  
  if(length(levels(as.factor(ctwas_gene_res$type))) == 1){
    ctwas_gene_res$gene_type <- G_list[sapply(ctwas_gene_res$id, match, G_list$ensembl_gene_id), "gene_biotype"]
    ctwas_gene_res$genename <- G_list[sapply(ctwas_gene_res$id, match, G_list$ensembl_gene_id), "hgnc_symbol"]
  } else {
    ctwas_gene_res$id_ens <- sub("\\|.*", "", ctwas_gene_res$id)
    ctwas_gene_res$gene_type <- G_list[sapply(ctwas_gene_res$id_ens, match, G_list$ensembl_gene_id), "gene_biotype"]
    ctwas_gene_res$genename <- G_list[sapply(ctwas_gene_res$id_ens, match, G_list$ensembl_gene_id), "hgnc_symbol"]
  }
  
  # Adding z values
  ctwas_gene_res$z <- z_gene$z[match(ctwas_gene_res$id, z_gene$id)]
  
  # Sorting and displaying datatable if required
  ctwas_gene_res <- ctwas_gene_res[order(-ctwas_gene_res$susie_pip),]
  if (display_datatable) {
    datatable(ctwas_gene_res[ctwas_gene_res$type != "SNP" & ctwas_gene_res$susie_pip > 0.8,])
  }
  
  return(ctwas_gene_res)
}

combine_pip <- function(ctwas_gene_res) {
  
  ctwas_gene_res <- ctwas_gene_res
    
  combined_pip <- ctwas_gene_res %>%
  filter(!is.na(genename)) %>% # Exclude rows where genename is NA
  group_by(genename) %>%
  summarise(
    total_susie_pip = sum(susie_pip, na.rm = TRUE), # Adding na.rm = TRUE to sum function
    pip_wingo_888_brain_expr_lasso = sum(susie_pip[type == "wingo_888_brain_expr_lasso"], na.rm = TRUE),
    pip_wingo_722_brain_protein_lasso = sum(susie_pip[type == "wingo_722_brain_protein_lasso"], na.rm = TRUE)
  )

  return(combined_pip)

}



```


# Full GWAS

## pqtl

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_fullgwas_mergef/output/AD_protein/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)
ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))

df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)

df_gene_pip08 <- df_gene[df_gene$susie_pip >0.8,]
DT::datatable(df_gene_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with susie pip >0.8'))

print("comparing with Wingo's study")

data_wingo <- read.table("/project2/guiming/xsun/proteome_alzheimer/1.brain_wingo/data/AD1_wingo.txt",header = T)
DT::datatable(data_wingo,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Wingo\'s results'))

data_overlap <- merge(df_gene_pip08, data_wingo, by.x="id", by.y = "Ensembl_gene_ID")
DT::datatable(data_overlap,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Overlapped'))
```

## joint analysis: eqtl & pqtl

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_fullgwas_mergef/output/AD_epjoint/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso,wingo_888_brain_expr_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)
ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))

df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)

combined_pip <- combine_pip(df_gene)
combined_pip08 <- combined_pip[combined_pip$total_susie_pip > 0.8,]
DT::datatable(combined_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8'))

df_gene_comb08 <- df_gene[df_gene$genename %in% combined_pip08$genename,]
DT::datatable(df_gene_comb08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Details for genes with combined pip > 0.8'))

print("comparing with Wingo's study")

data_wingo <- read.table("/project2/guiming/xsun/proteome_alzheimer/1.brain_wingo/data/AD1_wingo.txt",header = T)
DT::datatable(data_wingo,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Wingo\'s results'))

data_overlap <- merge(combined_pip08, data_wingo, by.x="genename", by.y = "Gene_symbol")
DT::datatable(data_overlap,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Overlapped'))
```

# Top locus removed

## pqtl

### results

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_topremoved_mergef/output/AD_protein/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)
ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))

df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)

df_gene_pip08 <- df_gene[df_gene$susie_pip >0.8,]
DT::datatable(df_gene_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with susie pip >0.8'))
```

### Comparing with Wingo's study

```{r }
data_wingo <- read.table("/project2/guiming/xsun/proteome_alzheimer/1.brain_wingo/data/AD1_wingo.txt",header = T)

susie_pip_wingo <- df_gene[df_gene$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol")

print(sprintf("There are %s genes reported by Wingo, %s of them are included in ctwas analysis, %s of them have susie pip > 0.8",nrow(data_wingo),nrow(susie_pip_wingo),sum(susie_pip_wingo$susie_pip >0.8)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al'))

```


## joint analysis: eqtl & pqtl

### Results 

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_topremoved_mergef/output/AD_epjoint/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso,wingo_888_brain_expr_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))

load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))

df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)

combined_pip <- combine_pip(df_gene)
combined_pip08 <- combined_pip[combined_pip$total_susie_pip > 0.8,]
DT::datatable(combined_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8'))

df_gene_comb08 <- df_gene[df_gene$genename %in% combined_pip08$genename,]
DT::datatable(df_gene_comb08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Details for genes with combined pip > 0.8'))
```

### Comparing with Wingo's study

```{r fig.height=10,fig.width=20}

susie_pip_wingo <- combined_pip[combined_pip$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol")

print(sprintf("There are %s genes reported by Wingo, %s of them are included in ctwas analysis, %s of them have combined susie pip > 0.8",nrow(data_wingo),nrow(susie_pip_wingo),sum(susie_pip_wingo$total_susie_pip >0.8)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al'))
```



## Compare the results for pqtl analysis and joint analysis

```{r}

df_overlap_p_ep <- combined_pip08[combined_pip08$genename %in% df_gene_pip08$genename,]     

print(sprintf("# of genes with susie pip > 0.8 (in pqtl analysis): %s",nrow(df_gene_pip08)))
print(sprintf("# of genes with combined susie pip > 0.8 (in joint analysis): %s",nrow(combined_pip08)))

print(sprintf("# of genes overlapped: %s",nrow(df_overlap_p_ep)))

df_uniq_ep <- combined_pip08[!combined_pip08$genename %in% df_gene_pip08$genename,]     

DT::datatable(df_uniq_ep,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Unique genes found by joint analysis'))

```

