---
title: "Jansen's GWAS (Merge = F)"
author: "XSun"
date: "2024-02-04"
output: 
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

# Overview

## Traits

Full GWAS (/project2/guiming/xsun/proteome_alzheimer/data_gwas/jansen.RDS) and top locus removed GWAS  (/project2/guiming/xsun/proteome_alzheimer/data_gwas/jansen.topremoved.RDS)

## Settings

1. preharmonize snp z score: harmonize_z=T, strand_ambig_action_z=“drop”

2. impute gene z-scores for both sets of prediction weights by chromosome harmonize_z = F (we have pre-harmonized) harmonize_wgt = T strand_ambig_action_wgt=“drop”;  scale_by_ld_variance=F (since the weight file was converted from FUSION format) 

3. ctwas_rss parameter estimation & fine mapping LD merge=F, group_prior_var_structure = “idependent”

4. Resourse requested: single qtl : --cpus-per-task=5 --mem=56G; joint analysis: --cpus-per-task=5 --mem=56G

5. Filtering using credible sets

## Functions used

```{r message=FALSE}
library(ctwas)
library(data.table)
library(DT)
library(dplyr)

thin <- 0.1
load("/project2/guiming/xsun/proteome_alzheimer/data_others/G_list_processed.RData")
G_list <- G_list[!duplicated(G_list$hgnc_symbol),]


process_para <- function(ctwas_parameters) {
  
  para <- cbind(ctwas_parameters$group_size,
                  ctwas_parameters$group_prior,
                  ctwas_parameters$group_prior_var,
                  c(NA, ctwas_parameters$enrichment),
                  ctwas_parameters$group_pve)
  para <- as.data.frame(cbind(rownames(para), para))
  colnames(para) <- c("group","group_size","group_prior","group_prior_var","enrichment","group_pve")
  return(para)
}

process_data <- function(weight, outputdir, outname, z_gene, display_datatable = TRUE) {
  
  if(!exists("G_list")) {
    load("/project2/guiming/xsun/proteome_alzheimer/data_others/G_list_processed.RData")
  }
  
  # Splitting and constructing weight file paths
  #weight <- unlist(strsplit(weight, split=","))
  #weight <- paste0("/project2/guiming/xsun/proteome_alzheimer/data_weights/", weight, ".db")

  # Reading cTWAS results
  ctwas_res <- fread(paste0(outputdir, outname, ".susieIrss.txt"))

  #separate gene and SNP results
  ctwas_gene_res <- ctwas_res[ctwas_res$type != "SNP", ]
  ctwas_gene_res <- data.frame(ctwas_gene_res)
  #####
  # ctwas_snp_res <- ctwas_res[ctwas_res$type == "SNP", ]
  # ctwas_snp_res <- data.frame(ctwas_snp_res)
  
  if(length(levels(as.factor(ctwas_gene_res$type))) == 1){
    ctwas_gene_res$gene_type <- G_list[sapply(ctwas_gene_res$id, match, G_list$ensembl_gene_id), "gene_biotype"]
    ctwas_gene_res$genename <- G_list[sapply(ctwas_gene_res$id, match, G_list$ensembl_gene_id), "hgnc_symbol"]
  } else {
    ctwas_gene_res$id_ens <- sub("\\|.*", "", ctwas_gene_res$id)
    ctwas_gene_res$gene_type <- G_list[sapply(ctwas_gene_res$id_ens, match, G_list$ensembl_gene_id), "gene_biotype"]
    ctwas_gene_res$genename <- G_list[sapply(ctwas_gene_res$id_ens, match, G_list$ensembl_gene_id), "hgnc_symbol"]
  }
  
  # Adding z values
  ctwas_gene_res$z <- z_gene$z[match(ctwas_gene_res$id, z_gene$id)]
  
  # Sorting and displaying datatable if required
  ctwas_gene_res <- ctwas_gene_res[order(-ctwas_gene_res$susie_pip),]
  if (display_datatable) {
    datatable(ctwas_gene_res[ctwas_gene_res$type != "SNP" & ctwas_gene_res$susie_pip > 0.8&ctwas_gene_res$cs_index!=0,])
  }
  
  return(ctwas_gene_res)
}

# combine_pip <- function(ctwas_gene_res) {
#   
#   ctwas_gene_res <- ctwas_gene_res
#     
#   combined_pip <- ctwas_gene_res %>%
#   dplyr::filter(!is.na(genename)) %>% # Exclude rows where genename is NA
#   group_by(genename) %>%
#   summarise(
#     total_susie_pip = sum(susie_pip, na.rm = TRUE), # Adding na.rm = TRUE to sum function
#     pip_wingo_888_brain_expr_lasso = sum(susie_pip[type == "wingo_888_brain_expr_lasso"], na.rm = TRUE),
#     pip_wingo_722_brain_protein_lasso = sum(susie_pip[type == "wingo_722_brain_protein_lasso"], na.rm = TRUE)
#   )
# 
#   return(combined_pip)
# 
# }

# combine_pip <- function(ctwas_gene_res) {
#   
#   # Summarize susie_pip by genename
#   susie_pip_summary <- ctwas_gene_res %>%
#     dplyr::filter(!is.na(genename)) %>%
#     dplyr::group_by(genename) %>%
#     dplyr::summarise(
#       total_susie_pip = sum(susie_pip, na.rm = TRUE),
#       pip_wingo_888_brain_expr_lasso = sum(susie_pip[type == "wingo_888_brain_expr_lasso"], na.rm = TRUE),
#       pip_wingo_722_brain_protein_lasso = sum(susie_pip[type == "wingo_722_brain_protein_lasso"], na.rm = TRUE),
#       .groups = 'drop'
#     )
#   
#   # Select the first occurrence of other attributes for each genename
#   first_occurrence_attributes <- ctwas_gene_res %>%
#     dplyr::arrange(genename, susie_pip) %>%
#     dplyr::distinct(genename, .keep_all = TRUE)
#   
#   # Merge the summaries with the first occurrence attributes
#   combined_pip <- dplyr::left_join(susie_pip_summary, first_occurrence_attributes, by = "genename") %>%
#     dplyr::select(genename, region_tag1, region_tag2, chrom, pos, cs_index, total_susie_pip, pip_wingo_888_brain_expr_lasso, pip_wingo_722_brain_protein_lasso)
#   
#   return(combined_pip)
# }

combine_pip <- function(ctwas_gene_res) {
  
  # Summarize susie_pip by genename
  susie_pip_summary <- ctwas_gene_res %>%
    dplyr::filter(!is.na(genename)) %>%
    dplyr::group_by(genename) %>%
    dplyr::summarise(
      total_susie_pip = sum(susie_pip, na.rm = TRUE),
      pip_wingo_888_brain_expr_lasso = sum(susie_pip[type == "wingo_888_brain_expr_lasso"], na.rm = TRUE),
      pip_wingo_722_brain_protein_lasso = sum(susie_pip[type == "wingo_722_brain_protein_lasso"], na.rm = TRUE),
      .groups = 'drop'
    )
  
  # Shuffle rows with the same genename and cs_index to handle ties randomly
  # Then, select the first occurrence of other attributes for each genename with the largest cs_index
  set.seed(123) # Ensure reproducibility for the random selection
  first_occurrence_attributes <- ctwas_gene_res %>%
    dplyr::mutate(random_order = sample(seq_len(n()))) %>%
    dplyr::arrange(genename, desc(cs_index), random_order) %>%
    dplyr::distinct(genename, .keep_all = TRUE)
  
  # Merge the summaries with the first occurrence attributes
  combined_pip <- dplyr::left_join(susie_pip_summary, first_occurrence_attributes, by = "genename") %>%
    dplyr::select(genename, region_tag1, region_tag2, chrom, pos, cs_index, total_susie_pip, pip_wingo_888_brain_expr_lasso, pip_wingo_722_brain_protein_lasso)
  
  return(combined_pip)
}


combine_pip_Cortex <- function(ctwas_gene_res) {
  
  ctwas_gene_res <- ctwas_gene_res
    
  combined_pip <- ctwas_gene_res %>%
  dplyr::filter(!is.na(genename)) %>% # Exclude rows where genename is NA
  group_by(genename) %>%
  summarise(
    total_susie_pip = sum(susie_pip, na.rm = TRUE), # Adding na.rm = TRUE to sum function
    pip_wingo_888_brain_expr_lasso = sum(susie_pip[type == "wingo_888_brain_expr_lasso"], na.rm = TRUE),
    pip_wingo_722_brain_protein_lasso = sum(susie_pip[type == "wingo_722_brain_protein_lasso"], na.rm = TRUE),
    pip_mashr_Brain_Cortex = sum(susie_pip[type == "mashr_Brain_Cortex"], na.rm = TRUE)
  )

  return(combined_pip)

}


combine_pip_Cerebellum <- function(ctwas_gene_res) {
  
  ctwas_gene_res <- ctwas_gene_res
    
  combined_pip <- ctwas_gene_res %>%
  dplyr::filter(!is.na(genename)) %>% # Exclude rows where genename is NA
  group_by(genename) %>%
  summarise(
    total_susie_pip = sum(susie_pip, na.rm = TRUE), # Adding na.rm = TRUE to sum function
    pip_wingo_888_brain_expr_lasso = sum(susie_pip[type == "wingo_888_brain_expr_lasso"], na.rm = TRUE),
    pip_wingo_722_brain_protein_lasso = sum(susie_pip[type == "wingo_722_brain_protein_lasso"], na.rm = TRUE),
    pip_mashr_Brain_Cortex = sum(susie_pip[type == "mashr_Brain_Cerebellum"], na.rm = TRUE)
  )

  return(combined_pip)

}


plotPieChartWithProportions <- function(para) {
  # Extract group names and pve values
  names <- as.character(para$group)
  numbers <- as.numeric(para$group_pve)
  
  # Calculate proportions
  proportions <- round(numbers / sum(numbers) * 100, 2)
  
  # Create labels with proportions
  labels <- paste(names, "-", proportions, "%")
  
  # Create pie chart with labels showing proportions
  pie(numbers, labels = labels, main = "Group PVE Proportions")
}

```



# Top locus removed

## pqtl

### results

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_topremoved_mergef/output/AD_protein/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)
ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
plotPieChartWithProportions(para)
df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)

df_gene_pip08 <- df_gene[df_gene$susie_pip >0.8 & df_gene$cs_index !=0,]
DT::datatable(df_gene_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with susie pip >0.8 & in credible sets'))
```

### Comparing with Wingo's PWAS

```{r }
data_wingo <- read.table("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_pwas.txt",header = T)

susie_pip_wingo <- df_gene[df_gene$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol")

print(sprintf("There are %s genes reported by Wingo, %s of them are included in ctwas analysis, %s of them have susie pip > 0.8, %s of them are in credible sets",nrow(data_wingo),nrow(susie_pip_wingo),sum(susie_pip_wingo$susie_pip >0.8),sum(data_merge$cs_index!=0)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al'))

```

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/VKORC1_pqtl.jpg" width="150%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/pqtl_ld.png" width="150%">
</figure>

## joint analysis: eqtl & pqtl

### Results 

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_topremoved_mergef/output/AD_epjoint/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso,wingo_888_brain_expr_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))

load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
plotPieChartWithProportions(para)
df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)
genes_included <- df_gene$genename
#df_gene <- df_gene[df_gene$cs_index !=0,]

combined_pip <- combine_pip(df_gene)
combined_pip08 <- combined_pip[combined_pip$total_susie_pip > 0.8,]
DT::datatable(combined_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8 & in credible sets'))

df_gene_comb08 <- df_gene[df_gene$genename %in% combined_pip08$genename,]
DT::datatable(df_gene_comb08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Details for genes with combined pip > 0.8'))
```

### Comparing with Wingo's twas & pwas

```{r fig.height=10,fig.width=20}

data_wingo_pwas <- read.table("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_pwas.txt",header = T)
data_wingo_pwas <- data_wingo_pwas[,c("Gene_symbol","cis.trans","PWAS_Z","PWAS_p.value","PWAS_FDR_p.value","COLOC_PP4")]
data_wingo_pwas$method <- "pwas"
data_wingo_twas <- readxl::read_excel("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_twas.xlsx")
data_wingo_twas <- data_wingo_twas[,-ncol(data_wingo_twas)]
data_wingo_twas$method <- "twas"
colnames(data_wingo_twas) <- colnames(data_wingo_pwas)

data_wingo <- rbind(data_wingo_pwas,data_wingo_twas)

susie_pip_wingo <- combined_pip[combined_pip$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol",all.y = T)

data_merge_pwas <- data_merge[data_merge$method =="pwas",]
print(sprintf("There are %s genes reported by Wingo pwas, %s of them are included in ctwas analysis, %s of them have combined susie pip > 0.8",nrow(data_merge_pwas),sum(data_merge_pwas$genename%in%genes_included),sum(data_merge_pwas[complete.cases(data_merge_pwas),]$total_susie_pip >0.8)))

gene_wingo_all <- data_merge[!duplicated(data_merge$genename),]

print(sprintf("There are %s genes reported by Wingo pwas + twas, %s of them have combined susie pip > 0.8 & in credible sets",nrow(gene_wingo_all),sum(gene_wingo_all[complete.cases(gene_wingo_all),]$total_susie_pip >0.8)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al (both twas and pwas)'))

df_our_uniq <- combined_pip08[!combined_pip08$genename %in%data_merge$genename,]
DT::datatable(df_our_uniq,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with combined susie pip > 0.8 but were not reported by wingo et al'))


```

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/RIN3_epqtl.jpg" width="150%">
</figure>

  
<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/ICA1L_epqtl.jpg" width="150%">
</figure>


## Compare the results for pqtl analysis and e+p QTL joint analysis

```{r}

df_overlap_p_ep <- combined_pip08[combined_pip08$genename %in% df_gene_pip08$genename,]     

print(sprintf("# of genes with susie pip > 0.8 (in pqtl analysis): %s",nrow(df_gene_pip08)))
print(sprintf("# of genes with combined susie pip > 0.8 (in joint analysis): %s",nrow(combined_pip08)))

print(sprintf("# of genes overlapped: %s",nrow(df_overlap_p_ep)))

df_uniq_ep <- combined_pip08[!combined_pip08$genename %in% df_gene_pip08$genename,]     

DT::datatable(df_uniq_ep,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Unique genes found by joint analysis'))

```


<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/VKORC1_epqtl_eqtltrack.jpg" width="150%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/eqtl_ld.png" width="150%">
</figure>

## joint analysis: eqtl & pqtl & sqtl from cortex

### Results 

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_topremoved_mergef/output/AD_epsjoint_cortex/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso,wingo_888_brain_expr_lasso,mashr_Brain_Cortex"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))

load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
plotPieChartWithProportions(para)
df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)
genes_included <- df_gene$genename
df_gene <- df_gene[df_gene$cs_index !=0,]

intro_mapping <- fread("/project2/guiming/xsun/proteome_alzheimer/data_others/intron_id_mapping/Brain_Cortex.leafcutter.phenotype_groups.txt.gz")
intro_mapping <- intro_mapping[intro_mapping$intron_id %in% df_gene$id_ens,]
intro_mapping$gene_id <- sub("(.*?)\\..*", "\\1", intro_mapping$gene_id)
intro_mapping$genename <- G_list[sapply(intro_mapping$gene_id, match, G_list$ensembl_gene_id), "hgnc_symbol"]
intro_mapping <- intro_mapping %>%
  mutate(match_id = sapply(strsplit(as.character(gtex_intron_id), ":"), function(x) x[length(x)]))
df_gene_updated <- df_gene %>%
  left_join(intro_mapping, by = c("id_ens" = "intron_id"))

# Step 2: Fill in NA values in genename of df_gene with genename from intro_mapping
df_gene_updated <- df_gene_updated %>%
  mutate(genename = ifelse(is.na(genename.x), genename.y, genename.x)) %>%
  dplyr::select(-genename.y, -matches("^intro_mapping"))

# Replace df_gene with updated data
df_gene <- df_gene_updated[,c("chrom","id","pos","type","region_tag1","region_tag2","cs_index","susie_pip","id_ens","z","genename")]


combined_pip <- combine_pip_Cortex(df_gene)
combined_pip08 <- combined_pip[combined_pip$total_susie_pip > 0.8,]
DT::datatable(combined_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8 & in credible sets'))

df_gene_comb08 <- df_gene[df_gene$genename %in% combined_pip08$genename,]
DT::datatable(df_gene_comb08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Details for genes with combined pip > 0.8'))
```

### Comparing with Wingo's twas & pwas

```{r fig.height=10,fig.width=20}

data_wingo_pwas <- read.table("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_pwas.txt",header = T)
data_wingo_pwas <- data_wingo_pwas[,c("Gene_symbol","cis.trans","PWAS_Z","PWAS_p.value","PWAS_FDR_p.value","COLOC_PP4")]
data_wingo_pwas$method <- "pwas"
data_wingo_twas <- readxl::read_excel("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_twas.xlsx")
data_wingo_twas <- data_wingo_twas[,-ncol(data_wingo_twas)]
data_wingo_twas$method <- "twas"
colnames(data_wingo_twas) <- colnames(data_wingo_pwas)

data_wingo <- rbind(data_wingo_pwas,data_wingo_twas)

susie_pip_wingo <- combined_pip[combined_pip$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol",all.y = T)

data_merge_pwas <- data_merge[data_merge$method =="pwas",]
print(sprintf("There are %s genes reported by Wingo pwas, %s of them are included in ctwas analysis, %s of them have combined susie pip > 0.8",nrow(data_merge_pwas),sum(data_merge_pwas$genename%in%genes_included),sum(data_merge_pwas[complete.cases(data_merge_pwas),]$total_susie_pip >0.8)))

gene_wingo_all <- data_merge[!duplicated(data_merge$genename),]

print(sprintf("There are %s genes reported by Wingo pwas + twas, %s of them have combined susie pip > 0.8 & in credible sets",nrow(gene_wingo_all),sum(gene_wingo_all[complete.cases(gene_wingo_all),]$total_susie_pip >0.8)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al (both twas and pwas)'))

df_our_uniq <- combined_pip08[!combined_pip08$genename %in%data_merge$genename,]
DT::datatable(df_our_uniq,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with combined susie pip > 0.8 but were not reported by wingo et al'))




```




## joint analysis: eqtl & pqtl & sqtl from cerebellum

### Results 

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_topremoved_mergef/output/AD_epsjoint_cerebellum/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso,wingo_888_brain_expr_lasso,mashr_Brain_Cerebellum"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))

load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
plotPieChartWithProportions(para)
df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)
genes_included <- df_gene$genename
df_gene <- df_gene[df_gene$cs_index !=0,]

intro_mapping <- fread("/project2/guiming/xsun/proteome_alzheimer/data_others/intron_id_mapping/Brain_Cerebellum.leafcutter.phenotype_groups.txt.gz")
intro_mapping <- intro_mapping[intro_mapping$intron_id %in% df_gene$id_ens,]
intro_mapping$gene_id <- sub("(.*?)\\..*", "\\1", intro_mapping$gene_id)
intro_mapping$genename <- G_list[sapply(intro_mapping$gene_id, match, G_list$ensembl_gene_id), "hgnc_symbol"]
intro_mapping <- intro_mapping %>%
  mutate(match_id = sapply(strsplit(as.character(gtex_intron_id), ":"), function(x) x[length(x)]))
df_gene_updated <- df_gene %>%
  left_join(intro_mapping, by = c("id_ens" = "intron_id"))

# Step 2: Fill in NA values in genename of df_gene with genename from intro_mapping
df_gene_updated <- df_gene_updated %>%
  mutate(genename = ifelse(is.na(genename.x), genename.y, genename.x)) %>%
  dplyr::select(-genename.y, -matches("^intro_mapping"))

# Replace df_gene with updated data
df_gene <- df_gene_updated[,c("chrom","id","pos","type","region_tag1","region_tag2","cs_index","susie_pip","id_ens","z","genename")]


combined_pip <- combine_pip_Cerebellum(df_gene)
combined_pip08 <- combined_pip[combined_pip$total_susie_pip > 0.8,]
DT::datatable(combined_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8 & in credible sets'))

df_gene_comb08 <- df_gene[df_gene$genename %in% combined_pip08$genename,]
DT::datatable(df_gene_comb08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Details for genes with combined pip > 0.8'))
```

### Comparing with Wingo's twas & pwas

```{r fig.height=10,fig.width=20}

data_wingo_pwas <- read.table("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_pwas.txt",header = T)
data_wingo_pwas <- data_wingo_pwas[,c("Gene_symbol","cis.trans","PWAS_Z","PWAS_p.value","PWAS_FDR_p.value","COLOC_PP4")]
data_wingo_pwas$method <- "pwas"
data_wingo_twas <- readxl::read_excel("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_twas.xlsx")
data_wingo_twas <- data_wingo_twas[,-ncol(data_wingo_twas)]
data_wingo_twas$method <- "twas"
colnames(data_wingo_twas) <- colnames(data_wingo_pwas)

data_wingo <- rbind(data_wingo_pwas,data_wingo_twas)

susie_pip_wingo <- combined_pip[combined_pip$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol",all.y = T)

data_merge_pwas <- data_merge[data_merge$method =="pwas",]
print(sprintf("There are %s genes reported by Wingo pwas, %s of them are included in ctwas analysis, %s of them have combined susie pip > 0.8",nrow(data_merge_pwas),sum(data_merge_pwas$genename%in%genes_included),sum(data_merge_pwas[complete.cases(data_merge_pwas),]$total_susie_pip >0.8)))

gene_wingo_all <- data_merge[!duplicated(data_merge$genename),]

print(sprintf("There are %s genes reported by Wingo pwas + twas, %s of them have combined susie pip > 0.8 & in credible sets",nrow(gene_wingo_all),sum(gene_wingo_all[complete.cases(gene_wingo_all),]$total_susie_pip >0.8)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al (both twas and pwas)'))

df_our_uniq <- combined_pip08[!combined_pip08$genename %in%data_merge$genename,]
DT::datatable(df_our_uniq,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with combined susie pip > 0.8 but were not reported by wingo et al'))
```





## Venn diagrammes for comparison

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/pqtl_pwas.jpg" width="150%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/epqtl_tpwas_comapre.jpg" width="150%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/pqtl_epqtl_compare.jpg" width="150%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/venn7.jpg" width="150%">
</figure>

<figure class="half">
    <img src="https://github.com/xsun1229/proteome_alzheimer/raw/master/output/venn8.jpg" width="150%">
</figure>

# Top locus removed for parameter estimation, full gwas for fine-mapping

Got the same retults as the 'Top locus removed' setting.

# Full GWAS

## pqtl

### results

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_fullgwas_mergef/output/AD_protein/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))
load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)
ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
plotPieChartWithProportions(para)
df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)

df_gene_pip08 <- df_gene[df_gene$susie_pip >0.8 & df_gene$cs_index !=0,]
DT::datatable(df_gene_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with susie pip >0.8'))
```

### Comparing with Wingo's pwas

```{r }
data_wingo <- read.table("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_pwas.txt",header = T)

susie_pip_wingo <- df_gene[df_gene$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol")

print(sprintf("There are %s genes reported by Wingo, %s of them are included in ctwas analysis, %s of them have susie pip > 0.8, %s of them are in credible sets",nrow(data_wingo),nrow(susie_pip_wingo),sum(susie_pip_wingo$susie_pip >0.8),sum(data_merge$cs_index!=0)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al'))

```


## joint analysis: eqtl & pqtl

### Results 

```{r fig.height=10,fig.width=20}
outputdir <- "/project2/guiming/xsun/proteome_alzheimer/19.jansen_0204_topremoved_mergef/output/AD_epjoint/"
outname <- "AD_ctwas"
gwas_n <- 472868
weight <- "wingo_722_brain_protein_lasso,wingo_888_brain_expr_lasso"
  
  
load(paste0(outputdir,outname,"_z_snp.Rd"))

load(paste0(outputdir,outname,"_z_gene.Rd"))

ctwas_parameters <- ctwas:::ctwas_summarize_parameters(outputdir = outputdir,
                                               outname = outname,
                                               gwas_n = gwas_n,
                                               thin = thin)

ctwas_parameters$convergence_plot

para <- process_para(ctwas_parameters)

DT::datatable(para,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Parameters'))
plotPieChartWithProportions(para)
df_gene <- process_data(weight = weight,outputdir = outputdir,outname = outname, z_gene =z_gene,)
genes_included <- df_gene$genename
df_gene <- df_gene[df_gene$cs_index !=0,]

combined_pip <- combine_pip(df_gene)
combined_pip08 <- combined_pip[combined_pip$total_susie_pip > 0.8,]
DT::datatable(combined_pip08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Genes with combined pip > 0.8 & in credible sets'))

df_gene_comb08 <- df_gene[df_gene$genename %in% combined_pip08$genename,]
DT::datatable(df_gene_comb08,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Details for genes with combined pip > 0.8'))
```

### Comparing with Wingo's pwas & twas

```{r fig.height=10,fig.width=20}

data_wingo_pwas <- read.table("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_pwas.txt",header = T)
data_wingo_pwas <- data_wingo_pwas[,c("Gene_symbol","cis.trans","PWAS_Z","PWAS_p.value","PWAS_FDR_p.value","COLOC_PP4")]
data_wingo_pwas$method <- "pwas"
data_wingo_twas <- readxl::read_excel("/project2/guiming/xsun/proteome_alzheimer/data_others/AD1_wingo_twas.xlsx")
data_wingo_twas <- data_wingo_twas[,-ncol(data_wingo_twas)]
data_wingo_twas$method <- "twas"
colnames(data_wingo_twas) <- colnames(data_wingo_pwas)

data_wingo <- rbind(data_wingo_pwas,data_wingo_twas)

susie_pip_wingo <- combined_pip[combined_pip$genename %in%data_wingo$Gene_symbol,]
data_merge <- merge(susie_pip_wingo, data_wingo, by.x="genename", by.y = "Gene_symbol",all.y = T)

data_merge_pwas <- data_merge[data_merge$method =="pwas",]
print(sprintf("There are %s genes reported by Wingo pwas, %s of them are included in ctwas analysis, %s of them have combined susie pip > 0.8",nrow(data_merge_pwas),sum(data_merge_pwas$genename%in%genes_included),sum(data_merge_pwas[complete.cases(data_merge_pwas),]$total_susie_pip >0.8)))

gene_wingo_all <- data_merge[!duplicated(data_merge$genename),]

print(sprintf("There are %s genes reported by Wingo pwas + twas, %s of them have combined susie pip > 0.8 & in credible sets",nrow(gene_wingo_all),sum(gene_wingo_all[complete.cases(gene_wingo_all),]$total_susie_pip >0.8)))

DT::datatable(data_merge,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','ctwas results for the genes reported by Wingo et al (both twas and pwas)'))

df_our_uniq <- combined_pip08[!combined_pip08$genename %in%data_merge$genename,]
DT::datatable(df_our_uniq,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','genes with combined susie pip > 0.8 but were not reported by wingo et al'))
```
